fast64_internal/mk64/__init__.py

```diff
@@ -0,0 +1,152 @@
+import bpy
+from bpy.types import PropertyGroup
+from ..f3d.f3d_material import createF3DMat
+from ..f3d.f3d_gbi import get_F3D_GBI
+from ..f3d.f3d_parser import getImportData, importMeshC
+from ..panels import MK64_Panel
+from ..utility import prop_split
+from bpy.utils import register_class, unregister_class
+from ..utility import raisePluginError
+from .f3d_course_parser import MK64F3DContext, parseCourseVtx
+
+
+class MK64_Properties(PropertyGroup):
+    """Global MK64 Scene Properties found under scene.fast64.mk64"""
+
+    # Import Course DL
+    EnableRenderModeDefault: bpy.props.BoolProperty(name="Enable Render Mode by Default", default=True)
+
+    @staticmethod
+    def upgrade_changed_props():
+        pass
+
+
+class MK64_ImportCourseDL(bpy.types.Operator):
+    # set bl_ properties
+    bl_idname = "object.fast64_mk64_course_import_dl"
+    bl_label = "Import Course DL"
+    bl_options = {"REGISTER", "UNDO", "PRESET"}
+
+    # Called on demand (i.e. button press, menu item)
+    # Can also be called from operator search menu (Spacebar)
+    def execute(self, context):
+        obj = None
+        if context.mode != "OBJECT":
+            bpy.ops.object.mode_set(mode="OBJECT")
+
+        try:
+            name = context.scene.DLImportName
+            importPath = bpy.path.abspath(context.scene.DLImportPath)
+            basePath = bpy.path.abspath(context.scene.DLImportBasePath)
```

Comment:
What you could do is by default use a decomp path, then when the user picks custom path you can let them pick the base path, see how oot does this!

Reply:
if you see in my [PR](https://github.com/n64decomp/mk64/pull/577/files#diff-992dd5495a9934608b67bb4a81f0f6710245f4caf6b3d9a0c3cda9115b8799e4) I define the path of the decomp automatically, so I don't see really the utility and how you would know this path to the decomp ?